### ISSUE TITLE
Problem Generating Application on GitHub

### ISSUE THREAD

**Issue by farzad-sedaghatbin** [jhipster/jhipster-online#184]

**Comment by gmarziou** [comment]
Could you please provide a description?

**Comment by farzad-sedaghatbin** [comment]
> Could you please provide a description?

it is stall in Generating application in repository ....

**Comment by vishal423** [comment]
seems related to https://github.com/jhipster/jhipster-online/issues/183

**Comment by SudharakaP** [comment]
Interestingly as in the previous case I don't see anything in the logs. I've restarted jhonline and it seems to fix the issue but I wonder why this keep happening. :thinking:

**Comment by SudharakaP** [comment]
Only logs I see for user @farzad-sedaghatbin are,

<details>
<summary>Expand Logs</summary>

```json
[
 {
   "textPayload": "Jul  9 11:17:15 jhipster-app jhonline.war[23197]: #033[2m2020-07-09 11:17:15.344#033[0;39m #033[32m INFO#033[0;39m #033[35m23222#033[0;39m #033[2m---#033[0;39m #033[2m[  XNIO-1 task-7]#033[0;39m #033[36mi.g.jhipster.online.service.LogsService #033[0;39m #033[2m:#033[0;39m log `d96c9592-0c4c-48c8-9e2d-e74b412b2505` : Generating application in repository farzad-sedaghatbin/jhipster-sample-applicationdd",
   "insertId": "s1anfxg1ipzv3y",
   "resource": {
     "type": "gce_instance",
     "labels": {
       "instance_id": "2786710281550632359",
       "zone": "europe-west1-b",
       "project_id": "jhipster-online"
     }
   },
   "timestamp": "2020-07-09T11:17:15Z",
   "labels": {
     "compute.googleapis.com/resource_name": "jhipster-app"
   },
   "logName": "projects/jhipster-online/logs/syslog",
   "receiveTimestamp": "2020-07-09T11:17:21.182491967Z"
 },
 {
   "textPayload": "Jul  9 11:17:15 jhipster-app jhonline.war[23197]: #033[2m2020-07-09 11:17:15.336#033[0;39m #033[32m INFO#033[0;39m #033[35m23222#033[0;39m #033[2m---#033[0;39m #033[2m[  XNIO-1 task-7]#033[0;39m #033[36mi.g.j.online.web.rest.GeneratorResource #033[0;39m #033[2m:#033[0;39m Generating application on GitHub - .yo-rc.json: {\"generator-jhipster\":{\"applicationType\":\"monolith\",\"gitCompany\":\"\",\"baseName\":\"jhipsterSampleApplicationdd\",\"packageName\":\"com.mycompany.myapp\",\"packageFolder\":\"com/mycompany/myapp\",\"serverPort\":8080,\"serviceDiscoveryType\":false,\"authenticationType\":\"jwt\",\"uaaBaseName\":\"../uaa\",\"cacheProvider\":\"ehcache\",\"enableHibernateCache\":true,\"websocket\":false,\"databaseType\":\"sql\",\"devDatabaseType\":\"h2Disk\",\"prodDatabaseType\":\"mysql\",\"searchEngine\":false,\"enableSwaggerCodegen\":false,\"messageBroker\":false,\"buildTool\":\"maven\",\"useSass\":true,\"clientPackageManager\":\"npm\",\"testFrameworks\":[],\"enableTranslation\":true,\"nativeLanguage\":\"en\",\"languages\":[\"en\"],\"clientFramework\":\"angularX\",\"jhiPrefix\":\"jhi\"},\"git-provider\":\"GitHub\",\"git-company\":\"farzad-sedaghatbin\",\"repository-name\":\"jhipster-sample-applicationdd\"}",
   "insertId": "s1anfxg1ipzv3x",
   "resource": {
     "type": "gce_instance",
     "labels": {
       "zone": "europe-west1-b",
       "project_id": "jhipster-online",
       "instance_id": "2786710281550632359"
     }
   },
   "timestamp": "2020-07-09T11:17:15Z",
   "labels": {
     "compute.googleapis.com/resource_name": "jhipster-app"
   },
   "logName": "projects/jhipster-online/logs/syslog",
   "receiveTimestamp": "2020-07-09T11:17:21.182491967Z"
 },
 {
   "textPayload": "Jul  9 09:17:28 jhipster-app jhonline.war[23197]: #033[2m2020-07-09 09:17:28.449#033[0;39m #033[32m INFO#033[0;39m #033[35m23222#033[0;39m #033[2m---#033[0;39m #033[2m[  XNIO-1 task-2]#033[0;39m #033[36mi.g.jhipster.online.service.LogsService #033[0;39m #033[2m:#033[0;39m log `a542ea06-2171-4efb-8bb5-a93173736130` : Generating application in repository farzad-sedaghatbin/jhipster-sample-applicationdd",
   "insertId": "1ji2ssng3n9avag",
   "resource": {
     "type": "gce_instance",
     "labels": {
       "project_id": "jhipster-online",
       "zone": "europe-west1-b",
       "instance_id": "2786710281550632359"
     }
   },
   "timestamp": "2020-07-09T09:17:28Z",
   "labels": {
     "compute.googleapis.com/resource_name": "jhipster-app"
   },
   "logName": "projects/jhipster-online/logs/syslog",
   "receiveTimestamp": "2020-07-09T09:17:31.578624020Z"
 },
 {
   "textPayload": "Jul  9 09:17:28 jhipster-app jhonline.war[23197]: #033[2m2020-07-09 09:17:28.441#033[0;39m #033[32m INFO#033[0;39m #033[35m23222#033[0;39m #033[2m---#033[0;39m #033[2m[  XNIO-1 task-2]#033[0;39m #033[36mi.g.j.online.web.rest.GeneratorResource #033[0;39m #033[2m:#033[0;39m Generating application on GitHub - .yo-rc.json: {\"generator-jhipster\":{\"applicationType\":\"monolith\",\"gitCompany\":\"\",\"baseName\":\"jhipsterSampleApplicationdd\",\"packageName\":\"com.mycompany.myapp\",\"packageFolder\":\"com/mycompany/myapp\",\"serverPort\":8080,\"serviceDiscoveryType\":false,\"authenticationType\":\"jwt\",\"uaaBaseName\":\"../uaa\",\"cacheProvider\":\"ehcache\",\"enableHibernateCache\":true,\"websocket\":false,\"databaseType\":\"sql\",\"devDatabaseType\":\"h2Disk\",\"prodDatabaseType\":\"mysql\",\"searchEngine\":false,\"enableSwaggerCodegen\":false,\"messageBroker\":false,\"buildTool\":\"maven\",\"useSass\":true,\"clientPackageManager\":\"npm\",\"testFrameworks\":[],\"enableTranslation\":true,\"nativeLanguage\":\"en\",\"languages\":[\"en\"],\"clientFramework\":\"angularX\",\"jhiPrefix\":\"jhi\"},\"git-provider\":\"GitHub\",\"git-company\":\"farzad-sedaghatbin\",\"repository-name\":\"jhipster-sample-applicationdd\"}",
   "insertId": "1ji2ssng3n9avaf",
   "resource": {
     "type": "gce_instance",
     "labels": {
       "project_id": "jhipster-online",
       "instance_id": "2786710281550632359",
       "zone": "europe-west1-b"
     }
   },
   "timestamp": "2020-07-09T09:17:28Z",
   "labels": {
     "compute.googleapis.com/resource_name": "jhipster-app"
   },
   "logName": "projects/jhipster-online/logs/syslog",
   "receiveTimestamp": "2020-07-09T09:17:31.578624020Z"
 },
 {
   "textPayload": "193.176.84.16 - - [09/Jul/2020:09:17:25 +0000] \"GET /api/GitHub/companies/farzad-sedaghatbin/projects HTTP/1.1\" 200 1346 \"https://start.jhipster.tech/generate-application\" \"Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.116 Safari/537.36\"",
   "insertId": "1esczceg42lsn1w",
   "resource": {
     "type": "gce_instance",
     "labels": {
       "instance_id": "2786710281550632359",
       "zone": "europe-west1-b",
       "project_id": "jhipster-online"
     }
   },
   "timestamp": "2020-07-09T09:17:25.104560213Z",
   "labels": {
     "compute.googleapis.com/resource_name": "jhipster-app"
   },
   "logName": "projects/jhipster-online/logs/apache-access",
   "receiveTimestamp": "2020-07-09T09:17:26.552356630Z"
 },
 {
   "textPayload": "Jul  8 12:21:58 jhipster-app jhonline.war[23197]: #033[2m2020-07-08 12:21:58.885#033[0;39m #033[32m INFO#033[0;39m #033[35m23222#033[0;39m #033[2m---#033[0;39m #033[2m[ XNIO-1 task-12]#033[0;39m #033[36mi.g.jhipster.online.service.LogsService #033[0;39m #033[2m:#033[0;39m log `18cd6f94-ac47-40b3-9f69-a770164ad516` : Generating application in repository farzad-sedaghatbin/jhipster-sample-applicationd",
   "insertId": "12ah142fufwar0",
   "resource": {
     "type": "gce_instance",
     "labels": {
       "project_id": "jhipster-online",
       "instance_id": "2786710281550632359",
       "zone": "europe-west1-b"
     }
   },
   "timestamp": "2020-07-08T12:21:58Z",
   "labels": {
     "compute.googleapis.com/resource_name": "jhipster-app"
   },
   "logName": "projects/jhipster-online/logs/syslog",
   "receiveTimestamp": "2020-07-08T12:22:02.638005415Z"
 },
 {
   "textPayload": "Jul  8 12:21:58 jhipster-app jhonline.war[23197]: #033[2m2020-07-08 12:21:58.877#033[0;39m #033[32m INFO#033[0;39m #033[35m23222#033[0;39m #033[2m---#033[0;39m #033[2m[ XNIO-1 task-12]#033[0;39m #033[36mi.g.j.online.web.rest.GeneratorResource #033[0;39m #033[2m:#033[0;39m Generating application on GitHub - .yo-rc.json: {\"generator-jhipster\":{\"applicationType\":\"monolith\",\"gitCompany\":\"\",\"baseName\":\"jhipsterSampleApplicationd\",\"packageName\":\"com.mycompany.myapp\",\"packageFolder\":\"com/mycompany/myapp\",\"serverPort\":8080,\"serviceDiscoveryType\":false,\"authenticationType\":\"jwt\",\"uaaBaseName\":\"../uaa\",\"cacheProvider\":\"ehcache\",\"enableHibernateCache\":true,\"websocket\":false,\"databaseType\":\"sql\",\"devDatabaseType\":\"h2Disk\",\"prodDatabaseType\":\"mysql\",\"searchEngine\":false,\"enableSwaggerCodegen\":false,\"messageBroker\":false,\"buildTool\":\"maven\",\"useSass\":true,\"clientPackageManager\":\"npm\",\"testFrameworks\":[],\"enableTranslation\":true,\"nativeLanguage\":\"en\",\"languages\":[\"en\"],\"clientFramework\":\"angularX\",\"jhiPrefix\":\"jhi\"},\"git-provider\":\"GitHub\",\"git-company\":\"farzad-sedaghatbin\",\"repository-name\":\"jhipster-sample-applicationd\"}",
   "insertId": "12ah142fufwaqz",
   "resource": {
     "type": "gce_instance",
     "labels": {
       "instance_id": "2786710281550632359",
       "project_id": "jhipster-online",
       "zone": "europe-west1-b"
     }
   },
   "timestamp": "2020-07-08T12:21:58Z",
   "labels": {
     "compute.googleapis.com/resource_name": "jhipster-app"
   },
   "logName": "projects/jhipster-online/logs/syslog",
   "receiveTimestamp": "2020-07-08T12:22:02.638005415Z"
 },
 {
   "textPayload": "Jul  8 12:21:46 jhipster-app jhonline.war[23197]: #033[2m2020-07-08 12:21:46.913#033[0;39m #033[32m INFO#033[0;39m #033[35m23222#033[0;39m #033[2m---#033[0;39m #033[2m[  XNIO-1 task-7]#033[0;39m #033[36mi.g.jhipster.online.service.LogsService #033[0;39m #033[2m:#033[0;39m log `f8604a06-a1de-4ab2-8949-4445ec5579fc` : Generating application in repository farzad-sedaghatbin/jhipster-sample-applicationd",
   "insertId": "8vqca9g35b64ci",
   "resource": {
     "type": "gce_instance",
     "labels": {
       "zone": "europe-west1-b",
       "project_id": "jhipster-online",
       "instance_id": "2786710281550632359"
     }
   },
   "timestamp": "2020-07-08T12:21:46Z",
   "labels": {
     "compute.googleapis.com/resource_name": "jhipster-app"
   },
   "logName": "projects/jhipster-online/logs/syslog",
   "receiveTimestamp": "2020-07-08T12:21:47.641426704Z"
 },
 {
   "textPayload": "Jul  8 12:21:46 jhipster-app jhonline.war[23197]: #033[2m2020-07-08 12:21:46.906#033[0;39m #033[32m INFO#033[0;39m #033[35m23222#033[0;39m #033[2m---#033[0;39m #033[2m[  XNIO-1 task-7]#033[0;39m #033[36mi.g.j.online.web.rest.GeneratorResource #033[0;39m #033[2m:#033[0;39m Generating application on GitHub - .yo-rc.json: {\"generator-jhipster\":{\"applicationType\":\"monolith\",\"gitCompany\":\"\",\"baseName\":\"jhipsterSampleApplicationd\",\"packageName\":\"com.mycompany.myapp\",\"packageFolder\":\"com/mycompany/myapp\",\"serverPort\":8080,\"serviceDiscoveryType\":false,\"authenticationType\":\"jwt\",\"uaaBaseName\":\"../uaa\",\"cacheProvider\":\"ehcache\",\"enableHibernateCache\":true,\"websocket\":false,\"databaseType\":\"sql\",\"devDatabaseType\":\"h2Disk\",\"prodDatabaseType\":\"mysql\",\"searchEngine\":false,\"enableSwaggerCodegen\":false,\"messageBroker\":false,\"buildTool\":\"maven\",\"useSass\":true,\"clientPackageManager\":\"npm\",\"testFrameworks\":[],\"enableTranslation\":true,\"nativeLanguage\":\"en\",\"languages\":[\"en\"],\"clientFramework\":\"angularX\",\"jhiPrefix\":\"jhi\"},\"git-provider\":\"GitHub\",\"git-company\":\"farzad-sedaghatbin\",\"repository-name\":\"jhipster-sample-applicationd\"}",
   "insertId": "8vqca9g35b64ch",
   "resource": {
     "type": "gce_instance",
     "labels": {
       "zone": "europe-west1-b",
       "project_id": "jhipster-online",
       "instance_id": "2786710281550632359"
     }
   },
   "timestamp": "2020-07-08T12:21:46Z",
   "labels": {
     "compute.googleapis.com/resource_name": "jhipster-app"
   },
   "logName": "projects/jhipster-online/logs/syslog",
   "receiveTimestamp": "2020-07-08T12:21:47.641426704Z"
 },
 {
   "textPayload": "51.255.31.54 - - [08/Jul/2020:12:21:39 +0000] \"GET /api/GitHub/companies/farzad-sedaghatbin/projects HTTP/1.1\" 200 1346 \"https://start.jhipster.tech/generate-application\" \"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.116 Safari/537.36\"",
   "insertId": "1aco7wdfawbzk6",
   "resource": {
     "type": "gce_instance",
     "labels": {
       "project_id": "jhipster-online",
       "zone": "europe-west1-b",
       "instance_id": "2786710281550632359"
     }
   },
   "timestamp": "2020-07-08T12:21:39.090036338Z",
   "labels": {
     "compute.googleapis.com/resource_name": "jhipster-app"
   },
   "logName": "projects/jhipster-online/logs/apache-access",
   "receiveTimestamp": "2020-07-08T12:21:42.563778673Z"
 }
]
```
</details>

which seems to indicate that the method, [`createGitProviderRepository`](https://github.com/SudharakaP/jhipster-online/blob/3b703c6cf8733ee68d868cb5ef7cc82e77dafef0/src/main/java/io/github/jhipster/online/service/GithubService.java#L199) is never called. Wondering if this is something specific to the VM maybe killing the threads. :thinking:

**Comment by pascalgrimaud** [comment]
we should change the level of log, so we'll see more things

**Comment by gmarziou** [comment]
Did you restart the app because it was down? If so, could it be that it was killed by the OOM killer? This is what happened to my apps, nothing in application logs, the only way to see it was in system logs

**Comment by SudharakaP** [comment]
@gmarziou : Thanks. No I restarted it to just to get the thing working again. We do have the syslogs; I've PMed you on twitter so perhaps I can share them with you as well.

**Comment by farzad-sedaghatbin** [comment]
solved

**Comment by vishal423** [comment]
IMO, we should keep it open and investigate on the real issue.

**Comment by SudharakaP** [comment]
I've change the log level to `DEBUG` for class; `io.github.jhipster.online.service.GithubService` so that if this happens again we can get more info. :thinking:

**Comment by SudharakaP** [comment]
@gmarziou : I've fixed the following and assigned the PR to you for review. 

- Fix compiler warnings which lacks parameters

- Fix all NullPointerExceptions coming from the `YoRCDeserializer.deserialize` method. 

- Refactor `GitHubService.getSyncedUserFromGitProvider` method such that it was broken into several smaller functions. 

- Make sure the StopWatch instances are stopped even after and exception occurs. 

Please let me know your thoughts and if you see anything else that can be improved. Thanks. :smile:

**Comment by SudharakaP** [comment]
The GitHub generation seems stuck again. Again it doesn't seem like it's calling `createGitProviderRepository` at all. Also I notice that there's continuous repetition of requests going to endpoint `/api/generate-application/` until the user logs out. :thinking:

**Comment by SudharakaP** [comment]
So to reiterate what @Julien mentioned in one of the emails so that everyone has context, 

>I believe the real issue is a thread or scheduler issue. For example, we have an issue with the mail provider and it "eats" all the threads from our Spring scheduler. It already happened in the past but I thought I solved all of it.

So it seems that since we have re-enabled the emails at, https://github.com/jhipster/jhipster-online/pull/170 this might be the cause of this issue. 

I would like to give it a try to fix the thread issue rather than reverting it back to be honest. So will be investigating on this a bit more. :thinking:

**Comment by julien** [comment]
> So to reiterate what @julien mentioned in one of the emails
I didn't mention anything

**Comment by jdubois** [comment]
Oh sorry @julien - this was for me!

**Comment by SudharakaP** [comment]
> > So to reiterate what @julien mentioned in one of the emails
> 
> I didn't mention anything

@julien : Ops sorry, I've corrected it. Trying to do too many things at once. :smile:

**Comment by julien** [comment]
@SudharakaP no worries

**Comment by lnagy82** [comment]
Any news with this bug? I can't generate on GitHub nor GitLab, but i can download as zip.

**Comment by SudharakaP** [comment]
@lnagy82 : I am waiting for #186 to be reviewed and merged. After that I am thinking to do a deployment to see if that fixes it. If not I will try to either revert (#170) or do some alternative fix. In any case, I've restarted the service for now so you should be able to generate it.

**Comment by SudharakaP** [comment]
Hi everyone, I've just deployed v2.6.4 and I haven't seen this problem ever since we deployed the changes from #186 (which was included in v2.6.3). Now if you encounter this issue again please feel free to reopen a new issue and/or ping me. :smile:  

cc: @farzad-sedaghatbin @lnagy82 @jshingler
